components/omega/src/ocn/auxiliaryVars/WindForcingAuxVars.h

```diff
@@ -0,0 +1,46 @@
+#ifndef OMEGA_AUX_WIND_H
+#define OMEGA_AUX_WIND_H
+
+#include "DataTypes.h"
+#include "HorzMesh.h"
+#include "OmegaKokkos.h"
+
+#include <string>
+
+namespace OMEGA {
+
+class WindForcingAuxVars {
+ public:
+   Array1DReal NormalStressEdge;
+   Array1DReal ZonalStressCell;
+   Array1DReal MeridStressCell;
+
+   WindForcingAuxVars(const std::string &AuxStateSuffix, const HorzMesh *Mesh,
+                      int NVertLevels);
+
+   KOKKOS_FUNCTION void computeVarsOnEdge(int IEdge, int KChunk) const {
+      if (KChunk == 0) {
+         const int JCell0 = CellsOnEdge(IEdge, 0);
+         const int JCell1 = CellsOnEdge(IEdge, 1);
+         const Real ZonalStressEdge =
+             0.5_Real * (ZonalStressCell(JCell0) + ZonalStressCell(JCell1));
+         const Real MeridStressEdge =
+             0.5_Real * (MeridStressCell(JCell0) + MeridStressCell(JCell1));
+
+         NormalStressEdge(IEdge) =
+             Kokkos::cos(AngleEdge(IEdge)) * ZonalStressEdge +
+             Kokkos::sin(AngleEdge(IEdge)) * MeridStressEdge;
+      }
+   }
```

Comment:
Should this be a more general edge reconstruction routine located elsewhere?

Also, this version is the anisotropic one we've moved away from for MPAS-Ocean. See https://github.com/E3SM-Project/E3SM/pull/6917.

Reply:
I implemented the isotropic version and created a functor that can do both variants in `HorzOperators.h`. I added a namelist option to choose:
```yaml
  WindStress:
    InterpType: Isotropic

```